crates/re_space_view_time_series/src/legacy_visualizer_system.rs

```diff
@@ -152,80 +153,92 @@ impl LegacyTimeSeriesSystem {
-                            return;
+                    |_timeless, entry_range, (times, _, scalars, scatterings, colors, radii, labels)| {
+                        let times = times.range(entry_range.clone()).map(|(time, _)| time.as_i64());
+
+                        // Allocate all points.
+                        points = times.map(|time| PlotPoint {
+                            time,
+                            ..default_point.clone()
+                        }).collect_vec();
+
+                        // Fill in values.
+                        for (i, scalar) in scalars.range(entry_range.clone()).enumerate() {
+                            if scalar.len() > 1 {
+                                re_log::warn_once!("found a scalar batch -- those have no effect");
+                            }
+                            points[i].value = scalar.first().map_or(0.0, |s| s.0);
                         }
 
-                        for (scalar, scattered, color, radius, label) in itertools::izip!(
-                            scalars.iter(),
-                            MaybeCachedComponentData::iter_or_repeat_opt(&scatterings, scalars.len()),
-                            MaybeCachedComponentData::iter_or_repeat_opt(&colors, scalars.len()),
-                            MaybeCachedComponentData::iter_or_repeat_opt(&radii, scalars.len()),
-                            MaybeCachedComponentData::iter_or_repeat_opt(&labels, scalars.len()),
-                        ) {
-                            let mut point = PlotPoint {
-                                time: time.as_i64(),
-                                value: scalar.0,
-                                ..default_point.clone()
-                            };
-
-                            // Make it as clear as possible to the optimizer that some parameters
-                            // go completely unused as soon as overrides have been defined.
-
-                            if override_color.is_none() {
-                                if let Some(color) = color.map(|c| {
-                                    let [r,g,b,a] = c.to_array();
-                                    if a == 255 {
-                                        // Common-case optimization
-                                        re_renderer::Color32::from_rgb(r, g, b)
-                                    } else {
-                                        re_renderer::Color32::from_rgba_unmultiplied(r, g, b, a)
+                        // Fill in series kind -- if available _and_ not overridden.
+                        if override_scattered.is_none() {
+                            if let Some(scatterings) = scatterings {
+                                for (i, scattered) in scatterings.range(entry_range.clone()).enumerate() {
+                                    if i >= points.len() {
+                                        re_log::debug_once!("more scatterings attributes than points -- that's a bug");
```

Comment:
a bug in what - the data or the code? If so, what part of the code? Can we print something actionable here, like the entity path?

Reply:
here and below